### PR TITLE
[SPARK-43914][SQL] Assign names to the error class _LEGACY_ERROR_TEMP_[2433-2437]

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
The pr aims to assign names to the error class _LEGACY_ERROR_TEMP_[2433-2437].

### Why are the changes needed?
Improve the error framework.

### Does this PR introduce _any_ user-facing change?
'No'.

### How was this patch tested?
Exists test cases updated.

### CODE REVIEW

File: core/src/main/resources/error/error-classes.json [hunk 1]
```diff
@@ -1883,6 +1883,13 @@
     ],
     "sqlState" : "42K05"
   },
+  "RESOLVED_PLAN_HAVE_CONFLICTING_ATTRS" : {
```
Comment (MaxGekk) [inline core/src/main/resources/error/error-classes.json:1886]: Can you trigger this errors from user space? Could you add some tests, please.

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisErrorSuite.scala [hunk 2]
```diff
@@ -787,8 +787,70 @@ class AnalysisErrorSuite extends AnalysisTest {
...
+        "plan" -> "'Join Cross\n:- LocalRelation <empty>, [a#0]\n+- LocalRelation <empty>, [a#0]\n",
+        "conflictingAttributes" -> "\"a\""
+      )
+    )
+  }
+
+  test("error test for self-intersect") {
+    val intersect = Intersect(testRelation, testRelation, true)
+    val error = intercept[AnalysisException] {
+      SimpleAnalyzer.checkAnalysis(intersect)
```
Comment (beliefer) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisErrorSuite.scala:804]: @MaxGekk I created test cases use `SimpleAnalyzer.checkAnalysis` directly.

File: core/src/main/resources/error/error-classes.json (hunk 1 above)
Comment (beliefer) [reply core/src/main/resources/error/error-classes.json:1886]: OK

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisErrorSuite.scala (hunk 2 above)
Comment (beliefer) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisErrorSuite.scala:804]: @MaxGekk Do you have any suggestion ?

**Comment by beliefer** [comment]
ping @MaxGekk The GA failure is unrelate to this PR.

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisErrorSuite.scala (hunk 2 above)
Comment (MaxGekk) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisErrorSuite.scala:804]: I see but if you cannot trigger the error from user space, we should convert the error to an internal one.

Comment (MaxGekk) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisErrorSuite.scala:804]: @beliefer Could you try to trigger it by from SQL.

Comment (beliefer) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisErrorSuite.scala:804]: > @beliefer Could you try to trigger it by from SQL.

We can't trigger it by SQL. Let me convert it to internal error.

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisErrorSuite.scala [hunk 3]
```diff
@@ -781,11 +781,43 @@ class AnalysisErrorSuite extends AnalysisTest {
 
   test("error test for self-join") {
     val join = Join(testRelation, testRelation, Cross, None, JoinHint.NONE)
-    val error = intercept[AnalysisException] {
+    val error = intercept[SparkException] {
       SimpleAnalyzer.checkAnalysis(join)
     }
-    assert(error.message.contains("Failure when resolving conflicting references in Join"))
-    assert(error.message.contains("Conflicting attributes"))
+    assert(error.getMessage.contains("Failure when resolving conflicting references in Join"))
```
Comment (MaxGekk) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisErrorSuite.scala:787]: Internal errors have their own error class, could you use `checkError()`, please.

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisErrorSuite.scala [hunk 4]
```diff
@@ -781,11 +781,43 @@ class AnalysisErrorSuite extends AnalysisTest {
...
+    assert(error.getMessage.contains("Conflicting attributes"))
+  }
+
+  test("error test for self-intersect") {
+    val intersect = Intersect(testRelation, testRelation, true)
+    val error = intercept[SparkException] {
+      SimpleAnalyzer.checkAnalysis(intersect)
+    }
+    assert(error.getMessage.contains(
```
Comment (MaxGekk) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisErrorSuite.scala:796]: Please, use `checkError`

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisErrorSuite.scala (hunk 2 above)
Comment (MaxGekk) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisErrorSuite.scala:794]: cc @cloud-fan Just to-double check that errors are internal.

**Review by MaxGekk — CHANGES_REQUESTED** [review]

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisErrorSuite.scala (hunk 3 above)
Comment (beliefer) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisErrorSuite.scala:787]: I see.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala [hunk 5]
```diff
@@ -673,9 +673,11 @@ trait CheckAnalysis extends PredicateHelper with LookupCatalog with QueryErrorsB
             }
 
           case p @ Project(exprs, _) if containsMultipleGenerators(exprs) =>
-            p.failAnalysis(
-              errorClass = "_LEGACY_ERROR_TEMP_2433",
-              messageParameters = Map("sqlExprs" -> exprs.map(_.sql).mkString(",")))
+            throw SparkException.internalError(
+              msg = "Only a single table generating function is allowed in a SELECT clause, " +
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala:677]: is this really internal? does the analyzer split `select explode(...), explode(...) ...` into two Projects?

Comment (beliefer) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala:677]: Let me have a try.

**Comment by beliefer** [comment]
ping @MaxGekk Rebased.

**Review by MaxGekk — APPROVED** [review]
@cloud-fan Are you ok with the changes?

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @beliefer and @cloud-fan for review.

**Comment by beliefer** [comment]
@MaxGekk @cloud-fan Thank you for all.

**Pull request closed without merging**